### PR TITLE
Scripts/Spells: Nourish

### PR DESCRIPTION
**Pull request by Sorikoff**
**Changes proposed:**
Migrate spell `Nourish` to spell scripts system.

**Target branch(es):** 3.3.5

~~**Issues addressed:**~~

**Tests performed:** Seems to be working correct, however, additional testing of output numbers would be appreciated.

~~**Known issues and TODO list:**~~

### CODE REVIEW

**Comment by jackpoz** [comment]
@Sorikoff could you do the "additional testing of output numbers" too ?

**Comment by Sorikoff** [comment]
Just checked. Difference with/without this PR for Nourish is ~1%.

**Comment by Keader** [comment]
It should be changed in Launch hook, no?

**Comment by Sorikoff** [comment]
Well, from what I see most of `HandleHeal` spells are using `OnEffectHitTarget`.

**Comment by jackpoz** [comment]
did you 2 agree if the current PR is ok or if it has to be changed ?

**Comment by Sorikoff** [comment]
I think it works the same as before. I can re-check if required but later.

**Comment by Sorikoff** [comment]
So tested again (no items, 80 lvl, all talents and spells + Glyph of Nourish learned). Difference with/without PR (5 casts of Nourish when Regrowth active) is about 2% (crits still matter).

File: src/server/scripts/Spells/spell_druid.cpp [hunk 1]
```diff
@@ -967,6 +968,39 @@ class spell_dru_moonkin_form_passive : public AuraScript
...
+        // Glyph of Nourish
+        if (AuraEffect const* aurEff = caster->GetAuraEffect(SPELL_DRUID_GLYPH_OF_NOURISH, EFFECT_0))
+        {
+            uint32 auraCount = 0;
+            int32 heal = GetHitHeal();
+
+            Unit::AuraEffectList const& periodicHeals = GetHitUnit()->GetAuraEffectsByType(SPELL_AURA_PERIODIC_HEAL);
+            for (AuraEffect const* hot : periodicHeals)
+            {
+                if (caster->GetGUID() == hot->GetCasterGUID())
```
Comment (ccrs) [inline src/server/scripts/Spells/spell_druid.cpp]: I know that in the original impl there was none, but, should we actually filter here for druid spells only?

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**